File: Abeyonix_Frontend/src/types/address.ts
export interface UserAddressBase {
  address_type: string;
  address_line1: string;
  address_line2?: string | null;
  city: string;
  state_province: string;
  postal_code: string;
  country: string;
  is_default?: boolean;
}

export interface UserAddressCreate extends UserAddressBase {}

export interface UserAddressResponse extends UserAddressBase {
  address_id: number;
}


export interface UserAddressUpdate {
  address_type?: string;
  address_line1?: string;
  address_line2?: string | null;
  city?: string;
  state_province?: string;
  postal_code?: string;
  country?: string;
  is_default?: boolean;
}


export interface MessageResponse {
  message: string;
}

File: Abeyonix_Backend/app/schemas/address.py
from pydantic import BaseModel
from typing import Optional, List


class UserAddressBase(BaseModel):
    address_type: str
    address_line1: str
    address_line2: Optional[str] = None
    city: str
    state_province: str
    postal_code: str
    country: str
    is_default: bool = False


class UserAddressCreate(UserAddressBase):
    pass


class UserAddressUpdate(BaseModel):
    address_type: Optional[str] = None
    address_line1: Optional[str] = None
    address_line2: Optional[str] = None
    city: Optional[str] = None
    state_province: Optional[str] = None
    postal_code: Optional[str] = None
    country: Optional[str] = None
    is_default: Optional[bool] = None


class UserAddressResponse(UserAddressBase):
    address_id: int

    class Config:
        from_attributes = True
